### PR TITLE
Remove azimuth carrier and add flattening ramps for AGE

### PR DESCRIPTION
**Pull request by vbrancat**
**This PR has been successfully tested to compute the absolute geolocation for Rosamond. It is awaiting further development in COMPASS **.

This PR adds the options to:
1. Remove the azimuth carrier phase ramp (for CSLC-S1 only)
2. Add back the flattening phase ramp 

when analyzing the absolute geolocation performance of CSLC-S1 products.
At the moment, the implementation supports CSLC-S1 only and will include CSLC-NI (GSLC) in a later stage.

According to our analyses:
1. Azimuth carrier ramp needs to be added back to CSLC-S1 to have meaningful interferograms. Therefore, the ramp should be removed when computing AGE as it affects AGE performance
2. As per current baseline, CSLC-S1 (and NISAR GSLC) will be flattened. Flattening phase ramp needs to be added back to avoid aliasing prior to oversampling the peak for AGE determination

### CODE REVIEW

**Comment by vbrancat** [comment]
@seongsujeong @LiangJYu can you please review this PR? Thanks

File: src/compass/utils/age.py [hunk 1]
```diff
@@ -17,7 +17,7 @@
 
 def run(cslc_file, cr_file, csv_output_file=None, plot_age=False,
         correct_set=False, mission_id='S1', pol='VV', ovs_factor=128,
-        margin=32):
+        margin=32, carrier=True, unflatten=True):
```
Comment (LiangJYu) [inline src/compass/utils/age.py:20]:
```suggestion
        margin=32, apply_az_ramp=True, unflatten=True):
```
nit - rename for clarity and consistency with `run` and `main`.

File: src/compass/utils/age.py [hunk 2]
```diff
@@ -306,14 +338,75 @@ def find_peak(cslc_file, x_loc, y_loc, mission_id='S1',
...
+        NI: NISAR (default: S1)
+
+    Returns
+    -------
+    flatten_phase: np.ndarray, float64
+        Numpy array containing flattening phase
+    '''
+
+    if mission_id == 'S1':
+        rg_off_path = 'science/SENTINEL1/CSLC/grids/range_offset'
```
Comment (LiangJYu) [inline src/compass/utils/age.py:391]:
```suggestion
        rg_off_path = f'{DATA_PATH}/range_offset'
```
same as above

File: src/compass/utils/age.py [hunk 3]
```diff
@@ -573,6 +666,12 @@ def create_parser():
                           help='Padding margin around CR position detected in the geocoded SLC '
                                'image. Actual margin is 2*margin from left-to-right and from'
                                'top-to-bottom')
+    optional.add_argument('-r', '--azimuth-ramp', dest='az_ramp', default=True,
```
Comment (LiangJYu) [inline src/compass/utils/age.py:669]:
```suggestion
    optional.add_argument('-r', '--azimuth-ramp', dest='apply_az_ramp', default=True,
```
nit - rename by adding a verb for clarity

File: src/compass/utils/age.py [hunk 4]
```diff
@@ -267,6 +289,16 @@ def find_peak(cslc_file, x_loc, y_loc, mission_id='S1',
     arr = get_cslc(cslc_file,
                    mission_id=mission_id,
                    pol=pol)
+    # If True, remove azimuth carrier ramp
+    if carrier:
+        carrier_phase = get_carrier_phase(cslc_file, mission_id=mission_id)
+        arr *= np.exp(-1j*carrier_phase)
```
Comment (seongsujeong) [inline src/compass/utils/age.py:295]:
```suggestion
        arr *= np.exp(-1j * carrier_phase)
```

File: src/compass/utils/age.py [hunk 5]
```diff
@@ -267,6 +289,16 @@ def find_peak(cslc_file, x_loc, y_loc, mission_id='S1',
...
+
+    # If True, adds back flattening phase
+    if unflatten:
+        flatten_phase = get_flatten_phase(cslc_file, mission_id=mission_id)
+        arr *= np.exp(-1j*flatten_phase)
```
Comment (seongsujeong) [inline src/compass/utils/age.py:300]:
```suggestion
        arr *= np.exp(-1j * flatten_phase)
```

File: src/compass/utils/age.py [hunk 6]
```diff
@@ -306,14 +338,75 @@ def find_peak(cslc_file, x_loc, y_loc, mission_id='S1',
     return x_cr, y_cr, snr_cr_db
 
 
+def get_carrier_phase(cslc_file, mission_id ='S1'):
```
Comment (seongsujeong) [inline src/compass/utils/age.py:341]:
```suggestion
def get_carrier_phase(cslc_file, mission_id='S1'):
```

**Review by seongsujeong — APPROVED** [review]
Providing few nit picks about the spacing. Approving this PR because they should not hold back this PR from getting merged.

File: src/compass/utils/age.py [hunk 7]
```diff
@@ -306,14 +338,75 @@ def find_peak(cslc_file, x_loc, y_loc, mission_id='S1',
...
+        NI: NISAR (default: S1)
+
+    Returns
+    -------
+    carrier_phase: np.ndarray, float64
+        Numpy array containing azimuth carrier phase
+    '''
+
+    if mission_id == 'S1':
+        carrier_path = 'science/SENTINEL1/CSLC/grids/carrier_phase'
```
[the remark this answers is not in the document]
Comment (vbrancat) [reply src/compass/utils/age.py:361]: This has been automatically taken care of by merging the main branch (note to myself)

**Pull request merged**